### ISSUE TITLE
`aria-selected` not announced for listbox options, weird/broken behaviour for grids

### ISSUE THREAD

**Issue by patrickhlauke** [nvaccess/nvda#15198]
### Steps to reproduce:

Listbox options:

1. Open https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/
2. Open the select-only combobox
3. Navigate through the options
4. Go to the option that is currently `aria-selected="true"` - visually indicated with a tick icon

Grids:

1. Open https://vaadin.com/docs/latest/example?path=component/datepicker/date-picker-basic.ts
2. Open the datepicker
3. Tab into the datepicker and use cursor keys to navigate calendar cells
4. Select a date using `Space`
5. Navigate away from the selected gridcell, then back

### Actual behavior:

Listbox options:

The selected state for the option is not announced (despite the `aria-selected="true"` attribute in the underlying markup, and developer tools showing that the option is indeed "Selected: true`

Grid:

After selecting one date in the calendar, in Chrome the currently selected cell with `aria-selected="true"` is *not* announced as selected. In Firefox, after selecting one cell, *all* gridcells in the grid are announced as being "selected".

### Expected behavior:

In both cases, I'd expect the selected option or gridcell to be announced as "selected" (and, in the latter example, I'd expect all non-selected gridcells NOT be announced as "selected").

### NVDA logs, crash dumps and other attachments:

Video recording showing the issue in Chrome/NVDA and Firefox/NVDA.

0:00-0:15 listbox in Chrome
0:15-0:24 listbox in Firefox
0:24-0:42 grid in Chrome
0:42-0:56 grid in Firefox

https://github.com/nvaccess/nvda/assets/895831/53831434-923b-4dd2-aac5-fc3d83ffb295

#### NVDA installed/portable/running from source:

Portable.

#### NVDA version:

2023.1 (2023.1.0.27913)

#### Windows version:

Windows 11

#### Name and version of other software in use when reproducing the issue:

Chrome 115.0.5790.110 (Official Build) (64-bit)

Firefox 115.0.2 (64-bit)

#### Does the issue still occur after restarting your computer?

Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

This has always been an issue with NVDA as far as I remember

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes

**Comment by patrickhlauke** [comment]
Note that Chrome/JAWS also has a problem with options in listboxes https://github.com/FreedomScientific/standards-support/issues/733, and the grid problem also appears in Firefox/JAWS https://github.com/FreedomScientific/standards-support/issues/734

**Comment by zersiax** [comment]
I get questions about this pretty regularly, particularly the grid cells sound like something that might cause issues in modern interactive grids, which we see more and more these days. Are we blocked on looking into this?
